src/EventListener/DeserializeListener.php

```diff
@@ -31,12 +32,14 @@
     private $serializer;
     private $serializerContextBuilder;
     private $formats;
+    private $formatsProvider;
 
-    public function __construct(SerializerInterface $serializer, SerializerContextBuilderInterface $serializerContextBuilder, array $formats)
+    public function __construct(SerializerInterface $serializer, SerializerContextBuilderInterface $serializerContextBuilder, array $formats, FormatsProviderInterface $formatsProvider = null)
```

Comment:
Same here with the constructor. I just want to omit we add another 5 arguments to the constructor because it makes migration from 2 to 3 harder.